zerver/views/messages.py

```diff
@@ -755,10 +753,8 @@ def same_realm_zephyr_user(user_profile, email):
     except ValidationError:
         return False
 
-    domain = email_to_domain(email)
-
     return user_profile.realm.is_zephyr_mirror_realm and \
-        RealmAlias.objects.filter(realm=user_profile.realm, domain=domain).exists()
+        get_realm_by_email_domain(email) == user_profile.realm
 
```

Comment:
We can just leave this as is. `get_realm_by_email_domain` doesn't work when `settings.REALMS_HAVE_SUBDOMAINS` is True, and there are zephyr realms with that setting. We can put a comment here (and for `same_realm_irc_user` and `same_realm_jabber_user`) instead, like "Note that this assumes subdomains_allowed=False for all RealmAliases corresponding to these realms."